### PR TITLE
Add InSAR processing

### PR DESCRIPTION
**Pull request by jhkennedy**
### Added
* `hyp3-gamma-insar` [v2.2.1](https://github.com/ASFHyP3/hyp3-insar-gamma/blob/develop/CHANGELOG.md#221)
  has been merged into HyP3 GAMMA at `hyp3_gamma.insar` to provide InSAR processing
  of Sentinel-1. `hyp3_gamma` now provides these additional entrypoints:
  * `insar` provides the InSAR process interface to HyP3
  * `ifm_sentinel.py` is the InSAR (science code) processor
  * `interf_pwr_s1_lt_tops_proc.py` for coregistering Sentinel-1 SLC data and DEMs
  * `unwrapping_geocoding.py` for unwrapping and geocoding Sentinel-1 INSAR products from GAMMA

### Changed
* `hyp3-gamma` entrypoint now allows InSAR processing through the `++process insar` option

---

This bring in the hyp3-insar-gamma plugin into the unified hyp3-gamma plugin, with all relevant git history.
    
The hyp3-insar-gamma history was filtered and rewritten using [git-filter-repo](https://github.com/newren/git-filter-repo) and by running these commands:

```
git clone git@github.com:ASFHyP3/hyp3-insar-gamma.git
cd hyp3-insar-gamma
    
git filter-repo \
     --path hyp3_insar_gamma/ \
     --path tests/test_entrypoints.py \
     --path tests/test_ifm_sentinel.py \
    --path tests/test_main.py \
    --path-rename 'hyp3_insar_gamma':'hyp3_gamma/insar' \
    --path-rename 'tests/test_entrypoints.py':'tests/insar/test_entrypoints.py' \
    --path-rename 'tests/test_ifm_sentinel.py':'tests/insar/test_ifm_sentinel.py' \
    --path-rename 'tests/test_main.py':'tests/insar/test_main.py' \
    --tag-rename '':'insar-'
    
git remote add gamma git@github.com:ASFHyP3/hyp3-gamma.git
git fetch --all
git merge --no-ff --allow-unrelated-histories gamma/develop
git push -u gamma master:add-insar
```

**Notably, this history will have new commit hashes as it is a rewrite.**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,7 +6,21 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [PEP 440](https://www.python.org/dev/peps/pep-0440/)
 and uses [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-##[3.0.0](https://github.com/ASFHyP3/hyp3-gamma/compare/v2.4.1...v3.0.0)
+## [3.1.0](https://github.com/ASFHyP3/hyp3-gamma/compare/v3.0.0...v3.1.0)
+
+### Added
+* `hyp3-gamma-insar` [v2.2.1](https://github.com/ASFHyP3/hyp3-insar-gamma/blob/develop/CHANGELOG.md#221)
```
Comment (asjohnston-asf) [inline CHANGELOG.md:12]: hyp3-insar-gamma

File: tests/test_entrypoints.py [hunk 2]
```diff
@@ -29,3 +29,31 @@ def test_hyp3_sentinel(script_runner):
...
+
+def test_insar_passthrough(script_runner):
+    ret = script_runner.run(
+        'hyp3_gamma', '++process', 'insar', '--help')
+    assert ret.success
+    assert 'insar' in ret.stdout
+    assert '--include-los-displacement' in ret.stdout
+
+
+def test_insar_gamma(script_runner):
```
Comment (asjohnston-asf) [inline tests/test_entrypoints.py:42]: could be just `test_insar()`

File: tests/test_util.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+from hyp3_gamma import util
+
+
```
Comment (asjohnston-asf) [inline tests/test_util.py:3]: some of the other functions that were moved to `util` could deserve tests, but not a priority with this PR

Comment (jhkennedy) [reply tests/test_util.py:3]: Agreed -- I think there will be a refactoring push after this.

**Review by Jlrine2 — APPROVED** [review]

**Pull request merged**